src/memory/flat.rs

```diff
@@ -203,4 +203,16 @@ impl<R: Register> Memory for FlatMemory<R> {
         memset(&mut self[addr as usize..(addr + size) as usize], value);
         Ok(())
     }
+
+    fn load_bytes(&mut self, addr: u64, length: usize) -> Result<Bytes, Error> {
+        if length == 0 {
+            return Ok(Bytes::new());
+        }
+        if length + addr as usize > self.memory_size() {
```

Comment:
I warned that length may be a dynamic value given by script user, so you must check length + addr is no overflowing.